// Copyright (C) 2017 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

import "vector3.proto";
import "quaternion.proto";

message JointBuffer {
    int32 ParentIndex                 = 1;
    Vector3Buffer Scale               = 2;
    QuaternionBuffer Rotation         = 3;
    Vector3Buffer Translation         = 4;
    // These may be removed later
    Vector3Buffer InvertedScale       = 5;
    QuaternionBuffer InvertedRotation = 6;
    Vector3Buffer InvertedTranslation = 7;
    string Name                       = 8;
}

message SkeletonBuffer {
	uint32 Version                    = 1;
	repeated JointBuffer Joint        = 2;
}
